File: Easy/2016. Maximum Difference Between Increasing Elements/Python/main.py
from typing import List

class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        min_so_far = nums[0]
        max_diff = -1
        for num in range(1:):
            if num > min_so_far:
                max_diff = max(max_diff, num - min_so_far)
            else:
                min_so_far: num

        return max_diff
                

File: Easy/2016. Maximum Difference Between Increasing Elements/Rust/main.rs
impl Solution {
    pub fn maximum_difference(nums: Vec<i32>) -> i32 {
        let mut min_so_far = nums[0];
        let mut max_diff = -1;

        for &num in nums.iter().skip(1) {
            if num > min_so_far {
                max_diff = max_diff.max(num - min_so_far);
            } else {
                min_so_far = num;
            }
        }

        max_diff
    }
}

